samples/msal-node-samples/ElectronSystemBrowserTestApp/README.md

```diff
@@ -0,0 +1,214 @@
+
+```javascript
+ "use static";
+
+import auth from "../config/auth.json" assert { type: "json" };
+
+window.addEventListener("DOMContentLoaded", () =>  {
+    let redirectUri = auth.customURLScheme;
+
+    if (window.location.search) {
+        redirectUri += window.location.search;
+        window.history.pushState({}, document.title, "/");
+    }
+
+    document.getElementById("SignIn").onclick = () => {
+        window.location.href = redirectUri;
+    };
+});
+
+```
+
+### Deep linking
+
+To register the custom URL scheme and to link to the electron app instance, we will utilize the `setAsDefaultProtocolClient()` API provided by electron. For more information about using deep linking with electron, check [electron documentation](https://www.electronjs.org/docs/latest/tutorial/launch-app-from-url-in-another-ap)
+
+```typescript
+private static setDefaultProtocol(): void {
+        if (process.defaultApp) {
+            if (process.argv.length >= 2) {
+                Main.application.setAsDefaultProtocolClient(
+                    authConfig.customProtocol.name, // e.g msal4b0db8c2-9f26-4417-8bde-3f0e3656f8e
+                    process.execPath,
+                    [path.resolve(process.argv[1])]
+                );
+            } else {
+                Main.application.setAsDefaultProtocolClient(
+                    authConfig.customProtocol.name // e.g msal4b0db8c2-9f26-4417-8bde-3f0e3656f8e
+                );
+            }
+        }
+    }
+```
+
+Next, we tell the application how to handle events that invoke the application instance (such as navigation to the custom url scheme in browser). This code will differ between OSs.
+
+#### In Windows
+
+When the external custom URL scheme is clicked it will trigger the event `Main.application.on("second-instance", Main.onSecondInstance);` where we can fetch the authorization code from custom URL scheme:
+
+```typescript
+ private static onSecondInstance(
+        event: any,
+        commandLine: string[],
+        workingDirectory: string
+    ): void {   
+        // Someone tried to run a second instance, we should focus our window.
+        if (Main.mainWindow) {
+            if (Main.mainWindow.isMinimized()) Main.mainWindow.restore();
+            Main.mainWindow.focus();
+        }
+
+        const url = Main.getDeepLinkUrl(commandLine);
+
+        if (url) {
+            Main.getCodeFromDeepLinkUrl(url);
+        }
```

Comment:
I have a question for you, and don't know the answer. When is it appropriate to combine the if condition and the following action into one line? Is it only when the action is a return statement? Additionally, I want to know why you combined them into one line on line 168, but not line 174-176.

Reply:
You're correct; I should've sustained the same code style in the if statement in line 174 as in-line 168. Regarding your question, When is it appropriate to combine the if condition and the following action into one line? When you want to improve your code readability and make your code easier to understand.